hatrac/model/storage/amazons3.py

```diff
@@ -112,29 +92,51 @@ def __init__(self, config):
         """
         PooledS3BucketConnection.__init__(self, config)
 
+    def _map_name(self, name):
```

Comment:
Not sure if my previous comment got lost.  I don't understand this multi-bucket/map_name feature.  It contradicts the design intentions of the S3 backend which is to put an entire S3 namespace in one bucket and assume that a sufficiently privileged hatrac HTTP client can create arbitrary namespaces and store objects under them, starting with just an empty root namespace.

Reply:
These changes were made to be able to support multi-tenant deployments in a single Hatrac instance, be that ill-advised or not.